internal/pkg/cli/task_run.go

```diff
@@ -0,0 +1,242 @@
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *runTaskOpts) validateImageName() error {
+	valid, err := regexp.MatchString(`^\d+\.dkr\.ecr\.[a-z0-9\-]+.amazonaws.com/[a-z][a-z0-9\-]*$`, o.Image)
+	if err != nil {
+		return fmt.Errorf("validate image name: %w", err)
+	}
+
+	if !valid {
+		return errors.New("image name is malformed")
+	}
+
+	return nil
+}
+
+func (o *runTaskOpts) validateSubnetID() error {
+	valid, err := regexp.MatchString(`^subnet-[a-z0-9]+`, o.SubnetID)
+	if err != nil {
+		return fmt.Errorf("validate subnet id: %w", err)
+	}
+
+	if !valid {
+		return errors.New("subnet id is malformed")
+	}
+
+	return nil
+}
+
+func (o *runTaskOpts) validateSecurityGroupIDs() error {
+	for _, id := range o.SecurityGroupIDs {
+		valid, err := regexp.MatchString(`^sg-[a-z0-9]+`, id)
+		if err != nil {
+			return fmt.Errorf("validate security group ids: %w", err)
+		}
+
+		if !valid {
+			return errors.New("one or more malformed security group id(s)")
+		}
+	}
+	return nil
+}
+
+func (o *runTaskOpts) validateAppName() error {
+	if o.AppName() == "" {
+		if _, err := o.store.GetApplication(o.App); err != nil {
+			return fmt.Errorf("get application: %w", err)
+		}
+	}
+	return nil
+}
+
+func (o *runTaskOpts) validateEnvName() error {
+	if o.App != "" {
+		if _, err := o.store.GetEnvironment(o.App, o.Env); err != nil {
+			return fmt.Errorf("get environment: %w", err)
+		}
+	} else {
+		if o.AppName() == "" {
+			return errNoAppInWorkspace
+		}
+
+		if _, err := o.store.GetEnvironment(o.AppName(), o.Env); err != nil {
+			return fmt.Errorf("get environment %s: %w", o.Env, err)
+		}
+	}
+
+	return nil
+}
+
+// BuildTaskRunCmd build the command for running a new task
+func BuildTaskRunCmd() *cobra.Command {
+	vars := runTaskVars{
+		GlobalOpts: NewGlobalOpts(),
+	}
+	cmd := &cobra.Command{
+		Use:   "run",
+		Short: "Run a task",
+		Long:  `Run a task.`,
+		Example: `
+Run a task with default setting
+/code $ copilot task run
+Run a task to the "test" environment under the current workspace.
+/code $ copilot task run --env test
+Starts 4 tasks with 2GB memory, Runs a particular image.
+/code $ copilot task run --num 4 --memory 2048 --task-role frontend-exec-role
+`,
+		RunE: runCmdE(func(cmd *cobra.Command, args []string) error {
+			opts, err := newTaskRunOpts(vars)
+			if err != nil {
+				return err
+			}
+			if err := opts.Validate(); err != nil { // validate flags
+				return err
+			}
+			return nil
+		}),
+	}
+
+	cmd.Flags().Uint8Var(&vars.Num, numFlag, 1, numFlagDescription)
+	cmd.Flags().Uint16Var(&vars.CPU, cpuFlag, 256, cpuFlagDescription)
+	cmd.Flags().Uint16Var(&vars.Memory, memoryFlag, 512, memoryFlagDescription)
+
+	cmd.Flags().StringVar(&vars.Image, imageFlag, "", imageFlagDescription)
+	cmd.Flags().StringVar(&vars.DockerfilePath, dockerFileFlag, "", dockerFileFlagDescription)
+
+	cmd.Flags().StringVar(&vars.TaskRole, taskRoleFlag, "", taskRoleFlagDescription)
+
+	cmd.Flags().StringVar(&vars.App, envFlag, "", appFlagDescription)
```

Comment:
```suggestion
	cmd.Flags().StringVar(&vars.appName, appFlag, "", appFlagDescription)
```

Reply:
App and AppName are slightly different, you can pass in an App which is not the app of the workspace.